pkg/handler/v2/external_cluster/node.go

```diff
@@ -478,3 +488,167 @@ func parseNodeConditions(node corev1.Node) (reason string, message string) {
+		}
+		cluster, err := getCluster(ctx, userInfoGetter, clusterProvider, privilegedClusterProvider, project.Name, req.ClusterID)
+		if err != nil {
+			return nil, common.KubernetesErrorToHTTPError(err)
+		}
+
+		var machineDeployment apiv2.ExternalClusterMachineDeployment
+
+		cloud := cluster.Spec.CloudSpec
+		if cloud != nil {
+			secretKeySelector := provider.SecretKeySelectorValueFuncFactory(ctx, privilegedClusterProvider.GetMasterClient())
+
+			if cloud.EKS != nil {
+				np, err := getEKSNodePool(ctx, cluster, req.MachineDeploymentID, secretKeySelector, cloud, clusterProvider)
+				if err != nil {
+					return nil, common.KubernetesErrorToHTTPError(err)
+				}
+				machineDeployment = *np
+			}
+		}
+
+		return machineDeployment, nil
+	}
+}
+
+func PatchMachineDeploymentEndpoint(userInfoGetter provider.UserInfoGetter, projectProvider provider.ProjectProvider, privilegedProjectProvider provider.PrivilegedProjectProvider, clusterProvider provider.ExternalClusterProvider, privilegedClusterProvider provider.PrivilegedExternalClusterProvider, settingsProvider provider.SettingsProvider) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		if !AreExternalClustersEnabled(settingsProvider) {
+			return nil, errors.New(http.StatusForbidden, "external cluster functionality is disabled")
+		}
+
+		req := request.(patchMachineDeploymentReq)
+		if err := req.Validate(); err != nil {
+			return nil, errors.NewBadRequest(err.Error())
+		}
+
+		project, err := common.GetProject(ctx, userInfoGetter, projectProvider, privilegedProjectProvider, req.ProjectID, &provider.ProjectGetOptions{IncludeUninitialized: false})
+		if err != nil {
+			return nil, common.KubernetesErrorToHTTPError(err)
+		}
+		cluster, err := getCluster(ctx, userInfoGetter, clusterProvider, privilegedClusterProvider, project.Name, req.ClusterID)
+		if err != nil {
+			return nil, common.KubernetesErrorToHTTPError(err)
+		}
+
+		cloud := cluster.Spec.CloudSpec
+		if cloud != nil {
+
+			secretKeySelector := provider.SecretKeySelectorValueFuncFactory(ctx, privilegedClusterProvider.GetMasterClient())
+
+			if cloud.EKS != nil {
+				var mdToPatch *apiv1.NodeDeployment
+				md, err := getEKSNodePool(ctx, cluster, req.MachineDeploymentID, secretKeySelector, cloud, clusterProvider)
+				if err != nil {
+					return nil, err
+				}
+				mdToPatch = &md.NodeDeployment
+
+				existingMDJSON, err := json.Marshal(mdToPatch)
+				if err != nil {
+					return nil, errors.NewBadRequest("cannot decode existing md: %v", err)
+				}
+				patchedMDJSON, err := jsonpatch.MergePatch(existingMDJSON, req.Patch)
+				if err != nil {
+					return nil, errors.NewBadRequest("cannot patch md: %v, %v", err, req.Patch)
+				}
+				var patchedMD *apiv1.NodeDeployment
+				err = json.Unmarshal(patchedMDJSON, &patchedMD)
+				if err != nil {
+					return nil, errors.NewBadRequest("cannot decode patched md: %v", err)
+				}
```

Comment:
this whole block of lines 553 to 572 is common for all providers. You can extract it and put it above